### ISSUE TITLE
Implement code splitting and lazy loading

### ISSUE THREAD

**Issue by mrfelton** [LN-Zap/zap-desktop#1216]
## Detailed Description

We should implement code splitting and lazy loading in order to help with overall app performance .

## Context

 - https://reactjs.org/docs/code-splitting.html
 - https://webpack.js.org/guides/code-splitting/

**Comment by bolatovumar** [comment]
This is always a nice thing to have but I doubt that this will have much of a performance impact in an Electron app. Code splitting is really nice on the web because it allows you to load code depending on when you need it and it can result in a significant perf boost depending on the user's internet connection.

Since Electron apps ship with all the code required to run the app I doubt much of a performance increase would be observed, It would be nice to know if there is a way to test this hypothesis though.

**Comment by mrfelton** [comment]
@bolatovumar agreed. My understanding is that there would be little impact to the Electron app in terms of asset load times, but there should be some improvement in terms of the time that it takes the browser to parse the code.

This is definitely not a priority, but in the long term I'd like to get the code in a state where it can be run fully in a standard browser environment. In fact, we do already run a lot of the code in a normal browser environment in the Storybook and our main bundle is currently over 2mb. Webpack recommends each one to be no larger than 200k.
